### ISSUE TITLE
data on s3

### ISSUE THREAD

**Issue by tischi** [tischi/i2k-2020-s3-zarr-workshop#6]
@constantinpape 

Just checking: This exists on s3?

```
      <Key>prospr-myosin.ome.zarr</Key>
      <SigningRegion>us-west-2</SigningRegion>
      <ServiceEndpoint>https://s3.embl.de</ServiceEndpoint>
      <BucketName>i2k-2020</BucketName>
      <Authentication>Anonymous</Authentication>
```

**Comment by tischi** [comment]
Related: Does this file exist?

```
prospr-myosin.ome.zarr/setup0/.zattrs
```

**Comment by constantinpape** [comment]
First question: yes, it's on the s3:
```
pape@gpu6:~$ mc ls embl/i2k-2020/prospr-myosin.ome.zarr/
[2020-11-20 13:15:54 CET]    24B .zgroup
[2020-11-20 16:57:04 CET]     0B setup0/
```

Second question: no, the file you are looking for is one step deeper in the hierarchy:
```
pape@gpu6:~$ mc ls embl/i2k-2020/prospr-myosin.ome.zarr/setup0/timepoint0
[2020-11-20 13:15:54 CET]   918B .zattrs
[2020-11-20 13:15:54 CET]    24B .zgroup
[2020-11-20 16:58:08 CET]     0B s0/
[2020-11-20 16:58:08 CET]     0B s1/
[2020-11-20 16:58:08 CET]     0B s2/
[2020-11-20 16:58:08 CET]     0B s3/
```

**Comment by tischi** [comment]
I just checked and the bdv n5 files have an attributes.json on every level in the hierarchy also  next to `setup0` and `timepoint0`. Thus, the N5Reader is looking for them and crashes at this point because it does not find it.
Specifically, this file 
````
prospr-myosin.ome.zarr/setup0/.zattrs
```` 

should contain such metadata, specifying the `dataType`: 

```
{"downsamplingFactors":[[1,1,1]],"dataType":"uint8"}
```

which is requested by this function in the `N5Reader`:

```
private < T extends NativeType< T >, V extends Volatile< T > & NativeType< V > > SetupImgLoader< T, V > createSetupImgLoader( final int setupId ) throws IOException
	{
		final String pathName = getPathName( setupId ); 
		final DataType dataType = n5.getAttribute( pathName, DATA_TYPE_KEY, DataType.class );
	 ...
```

I don't know enough about `ome.zarr` to suggest a solution.
@joshmoore?

If the solution requires a change of the N5ZarrReader, please do it here: 
https://github.com/mobie/mobie-viewer-fiji/tree/s3zarr

**Comment by constantinpape** [comment]
> I just checked and the bdv n5 files have an attributes.json on every level in the hierarchy also next to `setup0` and `timepoint0`. Thus, the N5Reader is looking for them and crashes at this point because it does not find it.

Yes, that is different compared to the ome multiscale format, which has metadata only in the group containing the zarr arrays for the different scales, in our case in `setup0/timepoint0/.zattrs`.
So introducing additional metadata in `setup0/.zattrs` would run counter the idea behind the multiscale ome zarr format a bit.

Ideally, we would change the `createSetupImgLoader` so that it reads all the metadata from  `setup0/timepoint0`.
I think I know which lines to change, I will make a suggestion later.

**Comment by constantinpape** [comment]
@tischi I marked the changes necessary to read the ome.zarr multiscale format in a draft pull request:
https://github.com/mobie/mobie-viewer-fiji/pull/148

This is pseudocode, but I hope it helps to show which changes are necessary to make it work.
If you still have issues with this we could do a remote programming session next week, I think then we would be able to crack this relatively fast.

**Comment by tischi** [comment]
@constantinpape 
Thanks a lot for the pointers. Using them, I should be able to make it work. I'll let you know.
